--- /dev/null
+++ lilo-22.8-bgtlb/BGRESCUE
@@ -0,0 +1,37 @@
+#!/bin/sh
+# This program version was patched for the use with BG-Rescue Linux and
+# to be compiled against uClibc 0.9.31 using gcc-4.2.1.
+#
+# Statically linking is used. Stripping is done with "sstrip".
+#
+# The Makefile, configure and building system has been patched to optimize 
+# for size ("-Os -s") and to produce i386 binaries.
+#
+# The installation process has been redirected to /opt/RESCUE/*PROGRAMNAME*
+
+DEBIAN_DIFF=""
+
+[[ -f "$DEBIAN_DIFF" ]] && 
+! [[ -f "$DEBIAN_DIFF.APPLIED" ]] && 
+touch "$DEBIAN_DIFF.APPLIED" &&
+echo "Applying patch $DEBIAN_DIFF" &&
+{ :
+! [[ `basename "$DEBIAN_DIFF" .diff` == "$DEBIAN_DIFF" ]] && cat "$DEBIAN_DIFF" | patch -p1 
+! [[ `basename "$DEBIAN_DIFF" .patch` == "$DEBIAN_DIFF" ]] && cat "$DEBIAN_DIFF" | patch -p1 
+! [[ `basename "$DEBIAN_DIFF" .tar` == "$DEBIAN_DIFF" ]] && tar -xvf "$DEBIAN_DIFF"
+
+if [[ -e debian/patches/00list ]]; 
+    then cat debian/patches/00list | awk '{print "cat ./debian/patches/" $0 "* | patch -p1" }' | sh 
+    elif [[ -e debian/patches/series ]]; 
+        then cat debian/patches/series | awk '{print "cat ./debian/patches/" $0 "* | patch -p1" }' | sh 
+        else for PATCH in `ls -1 debian/patches/*.patch debian/patches/*.diff debian/patches/*.dpatch 2>/dev/null`; do cat $PATCH | patch -p1; done 
+fi }
+
+make alles &&
+sstrip lilo lilo.static &&
+echo "Complete..." &&
+#echo -e "\nnow run:\tmake install-strip\n"
+#echo -e "\nnow run:\tsstrip  lilo\n" &&
+echo -e "\nnow run:\tcp  lilo.static  <<DESTINATION>>\n"
+#echo -e "\nnow run:\tBGRESCUE.INSTALL\n"
+#echo -e "Then run:\t sstrip /opt/RESCUE/<<PROGNAME>>/*bin/*\n"
--- lilo-22.8/Makefile
+++ lilo-22.8-bgtlb/Makefile
@@ -64,11 +64,14 @@
 #		  requested by VMware users (new, experimental)
 #   XL_SECS=n	  Support for extra large (non-standard) floppies.
 
-CONFIG=-DBDATA -DDSECS=3 -DEVMS -DIGNORECASE -DLVM -DNOKEYBOARD -DONE_SHOT \
-   -DPASS160 -DREISERFS -DREWRITE_TABLE -DSOLO_CHAIN -DVERSION -DVIRTUAL \
-   -DMDPRAID -DDEVMAPPER
-   
+#CONFIG=-DBDATA -DDSECS=3 -DEVMS -DIGNORECASE -DLVM -DNOKEYBOARD -DONE_SHOT \
+#   -DPASS160 -DREISERFS -DREWRITE_TABLE -DSOLO_CHAIN -DVERSION -DVIRTUAL \
+#   -DMDPRAID -DDEVMAPPER
 
+CONFIG=-DBDATA -DDSECS=3 -DIGNORECASE -DNOKEYBOARD -DONE_SHOT \
+   -DPASS160 -DREWRITE_TABLE -DSOLO_CHAIN -DVERSION \
+   -DMDPRAID
+
 # set the following if you wish LILO.COM to be installed
 DOS_DIR=/dosC/boot
 #
@@ -84,7 +87,7 @@
 
 # set the compiler optimization level
 
-OPT=-Os
+OPT=-Os -s
 
 ### End of configuration variables
 #
@@ -102,21 +105,22 @@
   sed 's/-D/-DLCF_/g'`
 LILO=-DLILO=0xbb920890
 KVER=`if [ -f /usr/include/linux/version.h ]; then echo -DHAS_VERSION_H; fi`
-HDM=`if [ -f /usr/include/libdevmapper.h ]; then echo -DHAS_LIBDEVMAPPER_H; fi`
-DEVMAPPER=`if [ -f /usr/include/libdevmapper.h ]; then echo "-ldevmapper"; fi`
+#HDM=`if [ -f /usr/include/libdevmapper.h ]; then echo -DHAS_LIBDEVMAPPER_H; fi`
+#DEVMAPPER=`if [ -f /usr/include/libdevmapper.h ]; then echo "-ldevmapper"; fi`
 
 TMP=_TmP_
 
 
 SHELL=/bin/sh
 CC=gcc
+STRIP=strip
 CPP=$(CC) -E
 AS86=as86 -0 -a
 LD86=ld86 -0
 NASM=nasm
 G=`cat foo1 foo2 | grep version | cut -d " " -f 3`
 
-CFLAGS=$(OPT) -Wall -g $(PCONFIG)
+CFLAGS=$(OPT) -Wall -g $(PCONFIG) -march=i386 -mtune=i386 -m32
 LDFLAGS=#-Xlinker -qmagic
 LIBS=$(DEVMAPPER)
 
@@ -359,6 +363,7 @@
 
 lilo:		$(OBJS)
 		$(CC) -o lilo $(LDFLAGS) $(OBJS) $(LIBS)
+		$(STRIP) -s lilo
 
 lilo.static:	$(OBJS)
 		$(CC) -o lilo.static -static $(LDFLAGS) $(OBJS) $(LIBS)
